recipes/zlib/all/conanfile.py

```diff
@@ -29,7 +29,11 @@ class ZlibConan(ConanFile):
 
     @property
     def _is_mingw(self):
-        return self.settings.os == "Windows" and self.settings.compiler == "gcc"
+        is_windows = self.settings.get_safe("os") == "Windows"
```

Comment:
This feels like a lot of "work" that will result in false in probably most cases (I don't have numbers but I assume mingw has quite a bit lower market share compared to gcc and msvc"
I guess it would still be better to get the short circuit behaviour compared to the old code where it returns false as soon as some condition is reached that makes it false.
I see later it is used for "if windows then _is_mingw()" but still thinks that it can do a lot lees most of the time.

Reply:
This a copy of conan-io/conan#12678 which still hasn't been merged. If you have any issues with it I recommend posting them there.

Reply:
Am I alright to resolve this?

Reply:
@seppeon I still think this is overly complex for a simple check. I would still recommend considering early return (see my comment [here](https://github.com/conan-io/conan/pull/12678#discussion_r1250971808)
I am also not a deciding voice so you can resolve if you don't intend to change 👍